openthread: fix not working ot diag repeat command

canisLupus1313 opened the pull request:
This commit fixes diag repeat command port by fixing issue with
incorrectly handled repeat timer.

Signed-off-by: Przemyslaw Bida <przemyslaw.bida@nordicsemi.no>

rlubos commented on subsys/net/lib/openthread/platform/alarm.c:
```diff
@@ -53,6 +53,12 @@ void platformAlarmProcess(otInstance *aInstance)
 	if (timer_ms_fired) {
 		timer_ms_fired = false;
 		otPlatAlarmMilliFired(aInstance);
```
Is it ok that the `otPlat*` function will be called twice if the diag is enabled? The bare metal implementeation does it slightly different:
https://github.com/openthread/ot-nrf528xx/blob/main/src/src/alarm.c#L500
Note the `else` condition.

canisLupus1313 replied:
You are right let's keep it similar.

rlubos reviewed (commented):
Please use the full name in the signoff entry.

Commit:
openthread: fix not working ot diag repeat command

This commit fixes diag repeat command port by fixing issue with
incorrectly handled repeat timer.

Signed-off-by: Przemyslaw Bida <przemyslaw.bida@nordicsemi.no>

rlubos commented on subsys/net/lib/openthread/platform/alarm.c:
```diff
@@ -52,7 +52,16 @@ void platformAlarmProcess(otInstance *aInstance)
 		timer_ms_fired = false;
-		otPlatAlarmMilliFired(aInstance);
+#if defined(CONFIG_OPENTHREAD_DIAG)
+		if (otPlatDiagModeGet())
+		{
```
```suggestion
		if (otPlatDiagModeGet()) {
```

rlubos commented on subsys/net/lib/openthread/platform/alarm.c:
```diff
@@ -52,7 +52,16 @@ void platformAlarmProcess(otInstance *aInstance)
+		if (otPlatDiagModeGet())
+		{
+			otPlatDiagAlarmFired(aInstance);
+		}
+		else
```
```suggestion
		} else
```

rlubos reviewed (commented):
Just to keep checkpatch happy (coding standard compliance).

rlubos commented on subsys/net/lib/openthread/platform/alarm.c:
```diff
@@ -52,7 +52,14 @@ void platformAlarmProcess(otInstance *aInstance)
-		otPlatAlarmMilliFired(aInstance);
+#if defined(CONFIG_OPENTHREAD_DIAG)
+		if (otPlatDiagModeGet()) {
+			otPlatDiagAlarmFired(aInstance);
```
It seems that `#include <openthread/platform/diag.h>` is missing, see failing CI.

canisLupus1313 replied:
Fixed.
